nautobot/extras/models/change_logging.py

```diff
@@ -181,3 +181,68 @@ def to_csv(self):
 
     def get_action_class(self):
         return ObjectChangeActionChoices.CSS_CLASSES.get(self.action)
+
+    def get_next_change(self, user=None):
+        """Return next change for this changed object, optionally restricting by user view permission"""
+        related_changes = self.get_related_changes().restrict(user, "view")
+        if user is not None:
+            related_changes = related_changes.restrict(user, "view")
+        return related_changes.filter(time__gt=self.time).order_by("time").first()
+
+    def get_prev_change(self, user=None):
+        """Return previous change for this changed object, optionally restricting by user view permission"""
+        related_changes = self.get_related_changes()
+        if user is not None:
+            related_changes = related_changes.restrict(user, "view")
```

Comment:
Maybe `user` could be an optional parameter to `get_related_changes` and have the restricting happen there, instead of needing to do `related_changes.restrict()` in each of `get_next_change`, `get_prev_change`, `get_extra_context`, etc?